[docs] clarify import rules for dynamic config (#28003)
expo/expo: docs/pages/workflow/configuration.mdx

For more customization, you can use the JavaScript (**app.config.js**) or [TypeScript](#using-typescript-for-configuration-appconfigts-instead-of) (**app.config.ts**). These configs have the following properties:

- Comments, variables, and single quotes.

> - `import` / `export` syntax is not supported, except when using [TypeScript with ts-node](https://docs.expo.dev/guides/typescript/#appconfigjs). JS files can be imported with `require()`, but they must be transpiled to support your current version of Node.js.

Reviewer: ```suggestion
- The `import` / `export` syntax is not supported, except when using [TypeScript with `ts-node`](/guides/typescript/#appconfigjs). JS files can be imported with `require()`. However, they must be transpiled to support your current version of Node.js.
```

Author: Reworked this all together, hopefully it's better

> - ESM import syntax (e.g., the `import` keyword) is not supported, except when using [TypeScript with ts-node](https://docs.expo.dev/guides/typescript/#appconfigjs). JS files that are compatible with your current version of Node.js can be imported with `require()`.

Author: oops, yes, I guess you could say `import` and `export` make up ESM, but not just when we're talking about imports ;-)

});
```

> To import other TypeScript files into **app.config.ts** or customize the language features, we recommend using [`ts-node`](/guides/typescript/#appconfigjs). `ts-node` also enables imports in any file imported by **app.config.ts**, meaning you can write local [config plugins](/config-plugins/introduction/) in TypeScript with full language features.

Reviewer: ```suggestion
To import other TypeScript files into **app.config.ts** or customize the language features, we recommend using [`ts-node`](/guides/typescript/#appconfigjs). `ts-node` enables imports in any file imported by **app.config.ts**. It means you can write local [config plugins](/config-plugins/introduction/) in TypeScript with full language features.
```
